cirq-core/cirq/circuits/circuit_operation.py

```diff
@@ -422,24 +478,22 @@ def dict_str(d: Dict) -> str:
             return circuit_msg
         return f'{circuit_msg}({", ".join(args)})'
 
-    def __hash__(self):
-        if self._hash is None:
-            object.__setattr__(
-                self,
-                '_hash',
-                hash(
-                    (
-                        self.circuit,
-                        self.repetitions,
-                        frozenset(self.qubit_map.items()),
-                        frozenset(self.measurement_key_map.items()),
-                        self.param_resolver,
-                        self.parent_path,
-                        tuple([] if self.repetition_ids is None else self.repetition_ids),
-                        self.use_repetition_ids,
-                    )
-                ),
+    @cached_property
+    def _hash(self):
```

Comment:
nit: add return type
```suggestion
    def _hash(self) -> int:
```